vm.go

```diff
@@ -53,6 +53,11 @@ type vmExt struct {
 	// isCode determines whether it should be evaluated as jsonnet code or
 	// treated as string.
 	isCode bool
+	// an ext or TLS code variable can also be initialized from an AST node
+	node ast.Node
+	// isNode determines if a code variable has been initialized using an AST node
+	// as opposed to a code string.
+	isNode bool
```

Comment:
Let's merge isCode and isNode into an enum and replace nested ifs with a single switch.